include/circt/Dialect/FIRRTL/FIRRTLStatements.td

```diff
@@ -52,6 +52,28 @@ def ConnectOp : FIRRTLOp<"connect", [FConnectLike]> {
   let hasCanonicalizer = true;
 }
 
+def ForeignOutputOp : FIRRTLOp<"foreign_output", [HasParent<"FModuleOp">, AllTypesMatch<["dest", "src"]>]> {
```

Comment:
Yes, because when implies when-sensitive things have implicit enables/muxing.

If we don't know what behavior a thing with these types have, we don't know what to do with it.

Since these cannot come in from fir files, this only affects injection of types during the pipeline. This can/should happen after lowering of whens (for those out-of-tree pipeline's sanity).

Reply:
Hopefully nothing declared (or defined) under when's needs to use this (e.g., plumbing out from an instance under a when), or they can have their own operation that lowers to the output operation before lower-to-hw, maybe?

Thanks for the explanations.

Reply:
Likewise, when lowering should either reject unknown ops, or leave the when with them still in it.